src/controller/CommissioningDelegate.h

```diff
@@ -573,6 +576,18 @@ class CommissioningParameters
     }
     void ClearICDStayActiveDurationMsec() { mICDStayActiveDurationMsec.ClearValue(); }
 
+    Span<const app::AttributePathParams> GetExtraReadPaths() const { return mExtraReadPaths; }
+
+    // Additional attribute paths to read as part of the kReadCommissioningInfo stage.
+    // These values read from the device will be available in ReadCommissioningInfo.attributes.
+    // Clients should avoid requesting paths that are already read internally by the commissioner
+    // as no consolidation of internally read and extra paths provided here will be performed.
+    CommissioningParameters & SetExtraReadPaths(Span<const app::AttributePathParams> paths)
```

Comment:
Should the dependency on CHIP_CONFIG_ENABLE_READ_CLIENT be documented?   Or maybe these APIs should just be conditioned on that?

Reply:
It’s that dependency mess again… all of commissioning actually depends on ReadClient but the header is parsed even when it’s disabled.

Reply:
Fair, but feels like we should just make sure if the logic backing the API is not present, then the API should not be present.

Reply:
None of CommissioningParameters is meaningfully usable without CHIP_CONFIG_ENABLE_READ_CLIENT... I don't really want go further down that path of pretending that CHIP_CONFIG_ENABLE_READ_CLIENT disables a select few methods within these classes... All of DeviceCommissioner and all of CommissioningDelegate.h should be behind the ifdef (or really in a separate header.) I can have a look at making the ifdef more coarse-grained like that as a separate PR. Maybe when folks are back after the break we can look into splitting CHIPDeviceController.{cpp, h} into two files in a git-history-preserving way and tidy this up a bit better that way.